metrics/serve.go

```diff
@@ -41,12 +41,15 @@ func registerMetrics(cfg Config) {
 	}
 }
 
-func cycleValues(cycleId int) {
+func cycleValues(cycleId *int, increment bool) {
```

Comment:
Let's put all related variables: metrics, and cycleId in one place and have functions operate on them directly.